website/docs/recipes/kafka.md

```diff
@@ -169,3 +170,217 @@ private MessageAndMetadata createPactRepresentationFor(Map<String, Object> metad
 In a similar fashion to the consumer test, we will need to get hold of a Kafka Serializer, which we should obtain through looking up the configuration of our production code (and again, in this demo, we'll just create an instance for use - make sure you do a better job!). 
 
 The `KafkaJsonSerializer` will take our domain object and turn it into a byte array. we pass this back to Pact which will compare it to what is held in the Pact broker. If its a match, we're green...
+
+# Schema Registry Compliant JSON
+The code examples here use Java, and are based on writing JSON messages to Kafka. If you are using AVRO or Protobuf, then the Kafka Schema-Registry examples may not yet be implemented (the PR for the JSON implementation can be found [here](https://github.com/pact-foundation/pact-jvm/pull/1503)).
+
+## Magic Bytes
+When a schema registry compliant message is written to a Kafka topic, the message (whether it is JSON, AVRO or Protobuf) is preceded by [5 "magic" bytes](https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#wire-format). If you were to use a compliant producer with a non-compliant consumer, things wouldn't work. We want our contract test to cover us here! Let's take a look at how this can be achieved (reminder - at the time of writing, this only works for JSON).
+
+## Schema Registry JSON Consumer
+We'll start with the bones of a class and we will use Junit5 annotations:
+```
+@ExtendWith(PactConsumerTestExt.class)
+@PactTestFor(providerName = "jsonSchemaKafkaProviderApp", providerType = ProviderType.ASYNCH, pactVersion = PactSpecVersion.V3)
+class JsonSchemaKafkaConsumerTest {
+...
+}
+```
+Next we define what we *expect* the contract to look like:
+```
+@Pact(consumer = "jsonSchemaKafkaConsumerApp")
+MessagePact schemaJsonPact(MessagePactBuilder builder) {
+    PactDslJsonBody body = new PactDslJsonBody();
+    body.stringType("name", "almost-anything2");
+
+    return builder.expectsToReceive("A json schema message")
+                  .withMetadata(Map.of("contentType", "application/vnd.schemaregistry.v1+json"))
+                  .withContent(body)
+                  .toPact();
+}
+```
+And there was our change: the metadata has been changed to match that of a schema registry compliant JSON message. This is significant - Pact will now be aware of the need for those 5 magic bytes mentioned earlier. We can now write our test. First, lets add a few constants and a field:
```

Comment:
would showing the snippet with and without the change, or having an annotated comment on the line, I assume it line 197 is the money ticket here from reading the paragraph on 202?

Reply:
Yes - I've given it a shot!